x/oracle/keeper/band_oracle_test.go

```diff
@@ -132,3 +133,118 @@ func TestBandCallDataRecord(t *testing.T) {
 	record = app.OracleKeeper.GetBandCallDataRecord(ctx, 1)
 	require.Nil(t, record)
 }
+
+func TestGetPrice(t *testing.T) {
+	app := app.Setup(t, false)
+	ctx := app.BaseApp.NewContextLegacy(false, tmproto.Header{Height: 1, ChainID: "3", Time: time.Unix(1618997040, 0)})
+
+	// Setup test data
+	bandPriceStateATOM := &types.BandPriceState{
+		Symbol:      "ATOM",
+		Rate:        math.NewInt(10),
+		ResolveTime: 1,
+		Request_ID:  1,
+		PriceState:  *types.NewPriceState(math.LegacyNewDec(10), 1),
+	}
+	bandPriceStateUSD := &types.BandPriceState{
+		Symbol:      "USD",
+		Rate:        math.NewInt(1),
+		ResolveTime: 1,
+		Request_ID:  1,
+		PriceState:  *types.NewPriceState(math.LegacyNewDec(1), 1),
+	}
+	invalidPriceStateATOM := &types.BandPriceState{
+		Symbol:      "ATOM",
+		Rate:        math.NewInt(0), // Invalid base rate
+		ResolveTime: 1,
+		Request_ID:  1,
+		PriceState:  *types.NewPriceState(math.LegacyNewDec(0), 1),
+	}
+
+	// Create variables for expected prices
+	expectedPrice10 := math.LegacyNewDec(10)
+	expectedPrice01 := math.LegacyNewDec(1).Quo(math.LegacyNewDec(10)) // 0.1
+
+	tests := []struct {
+		name            string
+		baseSymbol      string
+		quoteSymbol     string
+		basePriceState  *types.BandPriceState
+		quotePriceState *types.BandPriceState
+		expectedPrice   *math.LegacyDec
+		expectNil       bool
+	}{
+		// Fail cases first
+		{
+			name:            "Base, quote price do not exist, expect nil",
+			baseSymbol:      "ATOM",
+			quoteSymbol:     "USD",
+			basePriceState:  nil,
+			quotePriceState: nil,
+			expectedPrice:   nil,
+			expectNil:       true,
+		},
+		{
+			name:            "Base price is invalid (rate is zero), expect nil",
+			baseSymbol:      "ATOM",
+			quoteSymbol:     "USD",
+			basePriceState:  invalidPriceStateATOM,
+			quotePriceState: bandPriceStateUSD,
+			expectedPrice:   nil,
+			expectNil:       true,
+		},
+		// Pass cases
+		{
+			name:            "Valid base price (ATOM), quote does not exist, expect 10",
```

Comment:
With the name `quote does not exist` I think you should use another `quote` instead of `USD` because our logic will check if the `quote=USD` and then return the price of `base` after. I think we should separate into 2 cases, first pair ATOM/NOM and another ATOM/USD